src/components/AddCourse/AddForm/index.tsx

```diff
@@ -1,82 +1,102 @@
+  })
+  .map((option) => {
+    return <option value={option}>{option}</option>;
+  });
 
 const AddForm = (props: AddFormProps) => {
-    const [state, dispatch] = useContext(CoursesContext);
+  const [state, dispatch] = useContext(CoursesContext);
 
-    const [name, setName] = useState("");
-    const [code, setCode] = useState("");
-    const [credits, setCredits] = useState(0);
-    
-    const submitHandler = (event: any) => {
-        event.preventDefault();
-        const newCourse: Course = {
-            name,
-            prof: "John Doe",
-            strength: 0,
-            branch: event.target.course_branch.value,
-            year: event.target.course_year.value,
-            code,
-            credits
-        };
+  const [name, setName] = useState("");
+  const [code, setCode] = useState("");
+  const [credits, setCredits] = useState(0);
 
-        dispatch({type: 'ADD', payload: newCourse});
-        props.cancelHandler();
-    }
+  const submitHandler = (event: any) => {
+    event.preventDefault();
+    const newCourse: Course = {
+      name,
+      prof: "John Doe",
+      strength: 0,
+      branch: event.target.course_branch.value,
+      year: event.target.course_year.value,
+      code,
+      credits,
+    };
 
-    return (
-        <div className="form-container">
-           <form id="add-form" name="addCourse" onSubmit={submitHandler}> 
-                <MdClose className="close-form" onClick={props.cancelHandler}/>
-                <h2>Add new course</h2>
-                <label htmlFor="course_name">Course Name</label>
-                <input type="text" name="course_name" className="add-form-input" 
-                    value={name} onChange={(e) => setName(e.target.value)}/>
+    dispatch({ type: "ADD", payload: newCourse });
+    props.closeHandler();
+  };
 
-                <div className="form-small-input">
-                    <div>
-                        <label htmlFor="course_code">Course Code</label>
-                        <input type="text" name="course_code" className="add-form-input"
-                         value={code} onChange={(e) => setCode(e.target.value)}/>
-                    </div>
-                    <div>
-                        <label htmlFor="course_credits">Credits</label>
-                        <input type="number" name="course_credits" className="add-form-input"
-                         value={credits} onChange={(e) => setCredits(parseInt(e.target.value))}/>
-                    </div>
+  return (
+    <div className="form-container">
+      <form id="add-form" name="addCourse" onSubmit={submitHandler}>
+        <MdClose className="close-form" onClick={props.closeHandler} />
+        <h2>Add new course</h2>
+        <label htmlFor="course_name">Course Name</label>
+        <input
+          type="text"
+          name="course_name"
+          className="add-form-input"
+          value={name}
+          onChange={(e) => setName(e.target.value)}
```

Comment:
should be a single `handleOnChange` function handling changes for all the inputs.